apps/meteor/app/livechat/server/lib/Helper.ts

```diff
@@ -89,7 +90,7 @@ export const createLivechatRoom = async (
 	const { _id, username, token, department: departmentId, status = 'online' } = guest;
 	const newRoomAt = new Date();
 
-	const { activity } = guest;
+	const activity = guest.activity || (await LivechatContacts.findOneByVisitorId(guest._id, { projection: { activity: 1 } }))?.activity;
```

Comment:
```suggestion
	const activity = guest.activity || (await LivechatContacts.findOneByVisitorId<Pick<...>>(guest._id, { projection: { activity: 1 } }))?.activity;
```

if the func doesnt support, would be nice to add support